core/message-handling.go

```diff
@@ -253,14 +253,20 @@ func makeMessageStreamHandler(handle incomingMessageHandler, logger *logging.Log
 
 // startPeerConnections initiates asynchronous message exchange with
 // peer replicas.
-func startPeerConnections(replicaID, n uint32, connector api.ReplicaConnector, log messagelog.MessageLog, logger *logging.Logger) error {
-	supply := makePeerMessageSupplier(log)
-
+func startPeerConnections(replicaID, n uint32, connector api.ReplicaConnector, log messagelog.MessageLog, logger *logging.Logger, requestforward map[uint32]messagelog.MessageLog) error {
 	for peerID := uint32(0); peerID < n; peerID++ {
 		if peerID == replicaID {
 			continue
 		}
 
+		// TODO: using messagelog for buffering the forwarded requests could
+		// make the requests accumulated when the destination replica stops
+		// receiving them. Currently that does not happen because we only need
+		// to keep at most one request per client. We will need some update when
+		// we add support for concurrent client requests.
```

Comment:
Sorry for confusion, but indefinitely many Request messages can accumulate there, even though we currently don't support concurrent requests from the same client. This is because each _new_ Request may be added to the unicast message log for forwarding but never removed from there (check `core/internal/clientstate/request-seq.go`). 

I was thinking of a new component to replace the unicast message logs here. The new component would act as an asynchronous buffer, limited to one Request message per client. Maybe there are other possible solutions. Maybe we should update this comment, merge the PR and tackle the remaining problem as a separate issue.

Reply:
Thanks for the explanation, I updated the comment (pushed to the branch):

https://github.com/Naoya-Horiguchi/minbft/blob/683019e6dd07b1bb6d05ccb55b932fb17bbb24cc/core/message-handling.go#L262-L265

> I was thinking of a new component to replace the unicast message logs here. The new component would act as an asynchronous buffer, limited to one Request message per client. Maybe there are other possible solutions. 

I don't have any specific idea for now, so I'll look into "new component" approach first.